myhomework2.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+
+// не звертайте на цю функцію уваги
+// вона потрібна для того щоб правильно зчитати вхідні дані
+function readHttpLikeInput()
+{
+    $f = fopen('php://stdin', 'r');
+    $store = "";
+    $toread = 0;
+    while ($line = fgets($f)) {
+        $store .= preg_replace("/\r/", "", $line);
+        if (preg_match('/Content-Length: (\d+)/', $line, $m))
+            $toread = $m[1] * 1;
+        if ($line == "\r\n")
+            break;
+    }
+    if ($toread > 0)
+        $store .= fread($f, $toread);
+    return $store;
+}
+
+$contents = readHttpLikeInput();
+
+function parseTcpStringAsHttpRequest($string)
+{
+    return array(
+        "method" => getMethod($string),
+        "uri" => getUri($string),
+        "headers" => getHeaders($string),
+        "body" => getBody($string),
+    );
+}
+
+function getHeaders($string)
+{
+    $lines = explode("\n", $string);
+
+    $headers = array();
+
+    for ($i = 1; $i < sizeof($lines); $i++) {
+        if (strpos($lines[$i], ":")) {
+            $keyAndValue = explode(": ", $lines[$i]);
+            //$header[$keyAndValue[0]]= $keyAndValue[1];
+            $headers[] = $keyAndValue;
+        }
+    }
+
+    return $headers;
+}
+
+function getBody($string)
```

Comment:
you can do a tricky thing with explode to get body and headers separated, but with one line

Reply:
I understood that you mean that I can split through a blank line, but the condition said that the blank line is not always present

Reply:
I did it, but I had a question. It works only when I pass "\n\n" to explode but doesn't work when I use PHP_EOL . PHP_EOL. Is it because the line I am trying to split is written with /n and my system (Windows) expects \r\n when I use PHP_EOL?

Reply:
PHP_EOL is adapted for system, as far as I know.
https://www.php.net/manual/en/reserved.constants.php#constant.php-eol